tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -283,7 +298,15 @@ public int getFieldsSize() {
             return fields.size();
         }
 
-        public String format(NameObject nameObject) {
+        /**
+         * Return the rank order (0, 1, ...) in a list
+         * @return
+         */
+        public int getRank() {
+            return rank;
+        }
+
+        public String format(NameObject nameObject, ULocale formatterLocale, BreakIterator characterBreakIterator) {
```

Comment:
It seems unfortunate to require the break iterator to be passed in.  Couldn't we just instantiate it here if we need it?  For that matter, it also seems unfortunate that we need the locale.  It seems like we should be able to get that from the rules somehow, or at least store it in a property on the formatter instead of passing it to the format() method.

Reply:
For the breakIterator, I didn't want to pay the creation cost on every format call, and it seemed overkill to store it in every NamePattern.

Note: this does not have to be heavily optimized code, since the use will be in the CLDR tooling. However, I wanted to take a reasonable stab at what could make sense in a production environment.